eth/handler.go

```diff
@@ -162,7 +162,9 @@ func newHandler(config *handlerConfig) (*handler, error) {
 			log.Warn("Switch sync mode from full sync to snap sync")
 		}
 	} else {
-		if h.chain.CurrentBlock().Number.Uint64() > 0 {
+		blockNumber := h.chain.CurrentBlock().Number
+		// [Optimism] Allow snap sync from bedrock genesis block
+		if blockNumber.Uint64() > 0 && blockNumber.Cmp(config.Chain.Config().BedrockBlock) != 0 {
```

Comment:
```suggestion
		if blockNumber.Uint64() > 0 && (!config.Chain.Config().IsOptimism() || blockNumber.Cmp(config.Chain.Config().BedrockBlock) != 0) {
```

We have to check whether we are at optimism chain or not. Any better ideas? Code readability got worse, but it reduces op-geth diff.